services/ui-src/src/components/overlays/EntityDetailsDashboardOverlay.tsx

```diff
@@ -21,40 +21,20 @@ import {
 // assets
 import arrowLeftBlue from "assets/icons/icon_arrow_left_blue.png";
 // verbiage
-import { useStore } from "utils";
 
 export const EntityDetailsDashboardOverlay = ({
   closeEntityDetailsOverlay,
-  entityType,
   dashboard,
   selectedEntity,
   route,
 }: Props) => {
-  // Entity provider setup
-  const { clearEntities, setSelectedEntity, setEntityType } = useStore();
   const [selectedStep, setSelectedStep] = useState<
     OverlayModalPageShape | EntityDetailsOverlayShape
   >();
   const [stepIsOpen, setIsEntityStepOpen] = useState<boolean>(false);
-  const { report } = useStore();
 
   const { entitySteps } = route;
 
-  //using selectEntity directly will not re-render where there's a data change, we have to use its id to look up date from the report instead
-  const reportFieldDataEntities =
-    report?.fieldData[entityType!].find(
-      (entity: EntityShape) => entity.id === selectedEntity?.id
-    ) || [];
-
-  useEffect(() => {
-    setSelectedEntity(reportFieldDataEntities);
-    setEntityType(entityType);
-    return () => {
-      clearEntities();
-      setSelectedEntity(undefined);
-    };
-  }, [entityType, reportFieldDataEntities]);
-
```

Comment:
As part of the state management refactor, we've reduced the amount of times the selectedEntity is being set. Within the dashboard overlay, we no longer need to be setting or updating the selectedEntity, it should already have access to the updated entity from the store, by the time you enter this page.